frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-internal/raw-capacity-card/raw-capacity-card.tsx

```diff
@@ -10,20 +10,43 @@ import DashboardCardBody from '@console/shared/src/components/dashboard/dashboar
 import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
 import { FieldLevelHelp, humanizeBinaryBytes } from '@console/internal/components/utils';
 import { getInstantVectorStats } from '@console/internal/components/graphs/utils';
+import {
+  RedExclamationCircleIcon,
+  YellowExclamationTriangleIcon,
+} from '@console/shared/src/components/status/icons';
 import { CAPACITY_INFO_QUERIES } from '../../../../queries';
+import { DANGER_THRESHOLD, WARNING_THRESHOLD } from '../../common/capacity-breakdown/consts';
 import './raw-capacity-card.scss';
 
 const queries = (() => Object.values(CAPACITY_INFO_QUERIES))();
-const colorScale = ['#0166cc', '#d6d6d6'];
 
 // Enchance instantVectorStats to directly parse the values (else loading state won't be accurate)
 const parser = compose((val) => val?.[0]?.y, getInstantVectorStats);
 
+const CapacityStatusIcon: React.FC<CapacityStatusIconProps> = ({ ratio }) => {
+  const { t } = useTranslation();
+  if (ratio < WARNING_THRESHOLD) return null;
+  return (
+    <>
+      {ratio > DANGER_THRESHOLD && (
+        <RedExclamationCircleIcon title={t('ceph-storage-plugin~Error')} />
+      )}
+      {ratio > WARNING_THRESHOLD && ratio <= DANGER_THRESHOLD && (
+        <YellowExclamationTriangleIcon title={t('ceph-storage-plugin~Warning')} />
+      )}
+    </>
+  );
+};
+
 const RawCapacityCard: React.FC = React.memo(() => {
   const [values, loading] = usePrometheusQueries(queries, parser as any);
   const loadError = values.every((item) => typeof item === 'undefined' || item === 0);
   const { t } = useTranslation();
 
+  const generalColorScale = ['#0166cc', '#d6d6d6'];
+  const warningColorScale = ['#f0ab00', '#d6d6d6'];
+  const dangerColorScale = ['#c9190b', '#d6d6d6'];
```

Comment:
Keep it outside of the component. If you do that, you won't have to add it to the dependency list of useMemo.